metadata/src/utils/validation.rs

```diff
@@ -481,8 +557,19 @@ impl<'a> MetadataHasher<'a> {
 
         let extrinsic_hash = get_extrinsic_hash(&metadata.types, &metadata.extrinsic);
         let runtime_hash = get_type_hash(&metadata.types, metadata.runtime_ty(), &mut visited_ids);
```

Comment:
The `visited_ids` passed in here is only used in this one place anyway, which is good (see prev comment). Let's just pass in an `&mut HashSet::new()` to this function to ensure that we don't use it in some other place in the future and be consistent on "pass new visited_ids into each new top level hashing call".

Reply:
(looking around this file more I think we should re-think the `visited_ids` thing a bit or at least check that we're not overly using the same one; something I've probably done in other places too. I think ultimately we will want `get_type_hash` to _not_ take in `visited_ids` and instead construct it internally and then an internal version of `get_type_hash` can accept it for the recursive calls to it. Basically, each unique type that we hash should have start from a fresh set of visited IDs. I'll open an issue to do this more thoroughly)

Reply:
Opened https://github.com/paritytech/subxt/issues/1066 to generally fix/prevent misuse of `visited_type_ids`